### ISSUE TITLE
ionic upload with self signed cert

### ISSUE THREAD

**Issue by wedneyyuri** [ionic-team/ionic-cli#283]
I'm getting started with Ionic Framework and Ionic View App.

My steps:
1 - Install ionic using npm (already using last Cordova version)
`npm install -g ionic`

2 - Make a hello world app
`ionic start myApp sidemenu`

3 - Try to upload this app
**`ionic upload`**

I'm getting this error:
`Error logging in: Error:`**_SELF_SIGNED_CERT_IN_CHAIN (CLI v1.3.11)**_

```
Your system information:

OS: Windows 7 SP1
Node Version: v0.10.33
Cordova CLI: 4.3.0
Ionic Version: 1.0.0-beta.14
Ionic CLI Version: 1.3.11
```

My npm is set to **ignore strict-ssl**
`npm set strict-ssl false`

This issue is also on:
http://forum.ionicframework.com/t/ionic-upload-error-self-signed-cert-in-chain/18638

**Comment by jbavari** [comment]
@wedneyyuri Are you still having this issue? 

I might have a wild guess that was something to do with the login server cert at that time? 

Have you tried [this suggestion from the npm blog?](http://blog.npmjs.org/post/78165272245/more-help-with-self-signed-cert-in-chain-and-npm):

1) upgrade your version of npm

`npm install npm -g --ca=null`

2) tell your current version of npm to use known registrars

`npm config set ca=""`

**Comment by wedneyyuri** [comment]
Yes, I have already tried all these solutions but none worked. 

Only IONIC is giving errors, i'm using bower and it works fine.

**Comment by wedneyyuri** [comment]
Thanks for you reply @jbavari 

I believe that the problem is not related to the operating system, I got the same error on a Mac running Mavericks.

In the bower and npm I found settings to bypass the strict ssl ("strict-ssl": false), but have not seen anything like it in the ionic.

**Comment by ingshtrom** [comment]
I am also seeing this on Mac 10.10 as well with ionic CLI v1.3.20 and npm@2.7.3.

**Comment by samsur** [comment]
I am seeing the same error on Window 7 SP1
Node version : v0.12.2
Cordova CLI: 5.0.0
Ionic Version: 1.0.0-rc.4
Ionic CLI version: 1.3.20

**Comment by ingshtrom** [comment]
FWIW, what I found was that my company has us install specific certs (for whatever reason) so that we can access the internet. What I did to get around this is unplug my ethernet, switch to wifi, and go onto out public, "guest' wifi.  That doesn't require the certs to access the outside world, so I no longer got the error.  

TL;DR => Switch to public "guest" wifi for `ionic share` and `ionic upload` commands.

**Comment by wedneyyuri** [comment]
I can not use company computers outside the corporate network. 

Bower had a similar problem before. Solved on bower/bower@e09a3b8#diff-482e80f366fc4f68a917d0f2762e41d3R112.

They solved this problem by passing the strictSSL option to the "request" package (https://github.com/request/request#requestoptions-callback).

**Comment by sinedied** [comment]
As @wedneyyuri said, the simplest solution to this issue would be to add a --strict-ssl=false option like npm and bower, that will be passed to the `request` package.

**Comment by ethanturk** [comment]
We are having this issue, too. We had this with Bower and NPM and both have a strict-ssl config variable that will fix it. Ionic needs that option.

**Comment by LingboTang** [comment]
@sinedied @ethanturk I still got the save error when I have run `ionic push --ios-dev-cert --strict-ssl=false`

**Comment by sliultron** [comment]
@LingboTang I do not think ionic cli currently has the option --strict-ssl for either push and upload command

you can use the ionic help to check the existing options for commands
